### ISSUE TITLE
Improved statistics - installed vs. downloaded

### ISSUE THREAD

**Issue by roji** [NuGet/Home#931]
Apologies if this has been already discussed.

Current nuget package statistics show number of downloads, this may not be an optimal reflection of a package's popularity for several reasons:
* Automated tools (e.g. build server) download nugets as part of restore
* The same package may be deployed to many servers in the same organization, resulting in many logged downloads.

It seems that "number of installations" may be more useful than "number of downloads". This could be tracked at nuget.org by having nuget clients submit a special parameter when restoring a package as a result of an install. Clients could even inform nuget.org when *updating* packages. This could get nuget very close to actually tracking "unique installers".

Of course this doesn't imply that the current download count should be removed.

**Comment by yishaigalatzer** [comment]
We already track this data, see this link:

https://www.nuget.org/stats/packages/Microsoft.AspNet.Mvc?groupby=Version&groupby=ClientName&groupby=ClientVersion&groupby=Operation

but it turns out that it is not as meaningful for a few reasons.

1. Several key packages are installed by visual studio templates (see the example above) or from alternative/local servers.
2. Not all clients care to send the context information.
3. With NuGet 3 and DNX (ASP.NET 5) all dependencies are figured out during package restore, there is no difference between the two steps, which leads to more mixups of the data.

To summarize, we do agree the notion is good and we do track it. The result data didn't seem to be useful enough or meaningful enough. It is something we constantly discuss and try to improve on, and if we get to a point where the data is useful we will consider promoting it in the UI to be more visible.

**Comment by roji** [comment]
Thanks for the explanation @yishaigalatzer, those stats are what I had in mind and I understand the problems. Hopefully you manage to figure something out that improves on the existing download statistics...
